### PR TITLE
Add methods documentation to File class

### PR DESCRIPTION
**Pull request by adlerhsieh**

### CODE REVIEW

File: src/file.cr [hunk 1]
```diff
@@ -183,6 +192,15 @@ class File < IO::FileDescriptor
     File::Stat.new(stat).file?
   end
 
+  # Returns true if directory exists and is a directory
+  # 
+  # ```crystal
+  # # touch foo
+  # # mkdir bar
+  # File.file?("foo")    #=> false
```
Comment (maclover7) [inline src/file.cr]: Lines 200-202 show an example for `.file?`, not for `.directory?`

File: src/file.cr [hunk 2]
```diff
@@ -340,6 +395,12 @@ class File < IO::FileDescriptor
     lines
   end
 
+  # Write a file with a given filename and content.
+  # An existing file will be overwritten, and a non-existing file will be created.
+  #
+  # ```crystal
+  # File.write("./foo","bar")
```
Comment (maclover7) [inline src/file.cr]: A little nitpicky, but could you add a space between "./foo" and "bar"?

File: src/file.cr [hunk 3]
```diff
@@ -227,6 +254,12 @@ class File < IO::FileDescriptor
     end
   end
 
+  # Returns a file's extention. Returns an empty `String` if the file has no extension.
```
Comment (jhass) [inline src/file.cr]: "extension"

File: src/file.cr [hunk 4]
```diff
@@ -324,6 +364,13 @@ class File < IO::FileDescriptor
     end
   end
 
+  # Loop through each line in a file with a block of code
```
Comment (jhass) [inline src/file.cr]: "Yields each line of the given file to given block"?

File: src/file.cr [hunk 5]
```diff
@@ -386,6 +447,7 @@ class File < IO::FileDescriptor
     end
   end
 
+  # Returns the size of a given file with a `Int64` value.
```
Comment (jhass) [inline src/file.cr]: "Returns the size of the given file in bytes"?

File: src/file.cr [hunk 6]
```diff
@@ -332,6 +379,14 @@ class File < IO::FileDescriptor
     end
   end
 
+  # Returns an array of `String` that contains all lines from the file
```
Comment (jhass) [inline src/file.cr]: "Returns all lines of the given file as an array of strings"?

**Comment by jhass** [comment]
I think it's okay to refer to strings with "string"/"strings" instead of "\`String\`". That's also consistent with saying "an array of foo" etc.

File: src/file.cr (hunk 1 above)
Comment (adlerhsieh) [inline src/file.cr]: oops... I used copy and paste :P

File: src/file.cr (hunk 2 above)
Comment (adlerhsieh) [inline src/file.cr]: Yeap, sure

**Comment by adlerhsieh** [comment]
Yeah, sure, thanks for advices and corrections. Updated.

File: src/file.cr [hunk 7]
```diff
@@ -172,6 +172,15 @@ class File < IO::FileDescriptor
     LibC.access(filename, LibC::F_OK) == 0
   end
 
+  # Returns true if file exists and is a file
```
Comment (jhass) [inline src/file.cr]: Mmh, rereading this, the parameter is called "path", how about "Returns true if the given \*path\* exists and is a file."

File: src/file.cr [hunk 8]
```diff
@@ -183,6 +192,15 @@ class File < IO::FileDescriptor
     File::Stat.new(stat).file?
   end
 
+  # Returns true if directory exists and is a directory
```
Comment (jhass) [inline src/file.cr]: Ditto.

File: src/file.cr [hunk 9]
```diff
@@ -227,6 +254,12 @@ class File < IO::FileDescriptor
     end
   end
 
+  # Returns a file's extension. Returns an empty `String` if the file has no extension.
```
Comment (jhass) [inline src/file.cr]: "an empty string"

File: src/file.cr [hunk 10]
```diff
@@ -313,7 +346,14 @@ class File < IO::FileDescriptor
       file.close
     end
   end
-
+  
+  # Read the content from a file and returns a `String` value
```
Comment (jhass) [inline src/file.cr]: "Return the content of the given file as a string"?

File: src/file.cr [hunk 11]
```diff
@@ -324,6 +364,13 @@ class File < IO::FileDescriptor
     end
   end
 
+  # Yields each line of the given file to given block.
```
Comment (jhass) [inline src/file.cr]: "the given block"

File: src/file.cr [hunk 12]
```diff
@@ -340,6 +395,12 @@ class File < IO::FileDescriptor
     lines
   end
 
+  # Write a file with a given filename and content.
```
Comment (jhass) [inline src/file.cr]: "Write the given \*content\* to the given \*filename\*"

**Comment by adlerhsieh** [comment]
Updated with a bit fine-tuning.

**Comment by jhass** [comment]
Thanks

**Pull request merged**